homeassistant/components/shelly/__init__.py

```diff
@@ -142,5 +153,13 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
     )
     if unload_ok:
         await hass.data[DOMAIN].pop(entry.entry_id).shutdown()
+        if len(hass.data[DOMAIN]) == 2:
```

Comment:
This is fragile. We shouldn't reuse the dict for config entries and unsub listeners.